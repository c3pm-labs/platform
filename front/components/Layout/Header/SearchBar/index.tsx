import { Formik, Form } from 'formik';
import { useRouter } from 'next/router';
import { useTranslation } from 'next-i18next';

import TextInput from 'components/TextInput';

export interface SearchBarProps {
  className?: string;
  dataTestId?: string;
}

function SearchBar(props: SearchBarProps): JSX.Element {
  const {
    className,
    dataTestId,
  } = props;
  const router = useRouter();
  const { t } = useTranslation('common');
  const { q, tags } = router.query;
  const search = q || '';

  function handleSubmit(values: { search: string }): void {
    router.push({ pathname: '/search', query: { q: values.search, ...(tags ? { tags } : {}), page: 1 } });
  }

  return (
    <Formik
      onSubmit={handleSubmit}
      initialValues={{ search }}
      enableReinitialize
    >
      <Form
        className={className}
      >
        <TextInput
          data-testid={dataTestId}
          disableHelperText
          fullWidth
          name="search"
          type="search"
          placeholder={t('search')}
        />
      </Form>
    </Formik>
  );
}

export default SearchBar;
