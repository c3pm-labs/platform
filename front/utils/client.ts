import { ApolloClient, InMemoryCache, createHttpLink } from '@apollo/client'
import { setContext } from '@apollo/client/link/context'

const httpLink = createHttpLink({
  uri: process.env.GRAPHQL_URL,
  credentials: 'include',
})

const authLink = setContext((_, { headers }) => {
  return {
    headers: {
      ...headers,
    },
  }
})

const client = new ApolloClient({
  link: authLink.concat(httpLink),
  cache: new InMemoryCache(),
  ssrMode: typeof window === 'undefined',
})

export default client
