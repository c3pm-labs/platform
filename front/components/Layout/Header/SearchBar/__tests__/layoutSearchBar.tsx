import React from 'react';
import { MockedProvider } from '@apollo/client/testing';
import { render, fireEvent, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event'

import SearchBar from '../index';

// eslint-disable-next-line
const useRouter = jest.spyOn(require("next/router"), "useRouter");

test('Layout', () => {
  const mocks = [];

  useRouter.mockImplementation(() => ({
    route: '',
    pathname: '',
    query: { q: 'math' },
    asPath: '',
  }));

  const { container, getByTestId } = render(
    <MockedProvider mocks={mocks} addTypename={false}>
      <SearchBar />
    </MockedProvider>,
  );

  const input = screen.getByTestId('SearchBarInput');

  // userEvent.type(input, 'math');
  // expect(input).toHaveValue('math');
  fireEvent.submit(input);
  expect(container).toMatchSnapshot();
});
