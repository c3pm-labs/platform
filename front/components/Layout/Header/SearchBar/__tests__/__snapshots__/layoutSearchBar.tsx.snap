// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout 1`] = `
<div>
  <form
    action="#"
    data-testid="SearchBarInput"
  >
    <div
      class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
    >
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputTypeSearch MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
          id="search"
          name="search"
          placeholder="search..."
          type="search"
          value="math"
        />
        <fieldset
          aria-hidden="true"
          class="PrivateNotchedOutline-root-2 MuiOutlinedInput-notchedOutline"
          style="padding-left: 8px;"
        >
          <legend
            class="PrivateNotchedOutline-legend-3"
            style="width: 0.01px;"
          >
            <span>
              â€‹
            </span>
          </legend>
        </fieldset>
      </div>
      
    </div>
  </form>
</div>
`;
